### PR TITLE
fix(curriculum): corrected test and added solution for Create a Hash table challenge

### LINKED ISSUE #37577: Small bug in the Create a Hash table challenge
**Describe your problem and how to reproduce it:**
The Create a Hash table challenge has a bug in the 2nd to last test (see below):
```yml
  - text: Items are added using the hash function.
    testString: assert((function() { var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; called = 0; test.add('key1','value1'); test.add('key2','value2'); test.add('key3','value3'); return (called === 3)})());
  - text: The hash table handles collisions.
```
The problem is if the user's code calls the `hash` function more than once, then the the `called` variable will be greater than 3.  [This forum topic](https://www.freecodecamp.org/forum/t/having-trouble-with-create-a-hash-table-challenge/324284) shows an example of a solution that should technically pass all of the tests, but fails the 2nd to last test due to calling the `hash` function two times (once with the `hasOwnProperty` and again within the ternary expression.

**Proposed Solution:**
Since, we are trying to confirm there are `3` adds executed, we could just validate that `called` is greater or equal than `3` AND `called % 3` is equal to `0`.  

What does everyone think about my proposed solution?

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/create-a-hash-table

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #37577

This PR addresses a bug and adds a working solution for the challenge.

### CODE REVIEW

**Review by moT01 — COMMENTED** [review]
This PR solves the problem in the issue, and I'm fine with adding it like this. But I did notice a problem with the other tests. My remove method was failing because my lookup method was wrong.

The `add` and `remove` tests use the `lookup` method to confirm they are working. Well, they will fail even if they are correct unless the lookup method is also correct...
```
  - text: The add method adds key value pairs and the lookup method returns the values associated with a given key.
    testString: assert((function() { var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; test.add('key', 'value'); return (test.lookup('key') === 'value')})());
  - text: The remove method accepts a key as input and removes the associated key value pair.
    testString: assert((function() { var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; test.add('key', 'value'); test.remove('key'); return (test.lookup('key') === null)})());
```

Not sure if you want to try and take a look at this here @RandellDawson? Or just leave it for now, this is probably not an area a lot of campers make it to.

**Comment by RandellDawson** [comment]
>Not sure if you want to try and take a look at this here @RandellDawson? Or just leave it for now, this is probably not an area a lot of campers make it to.

@moT01 Could you add this other issue to a new GitHub issue so we don't lose track of it.  Since it is a completely separate issue from the issue this PR addresses, I would prefer to handle it on a new PR.  If you happen to know of a solution, feel free to add a commit.  Otherwise, can you approve and add this PR to the list of curriculum PRs?

Thanks.

**Review by moT01 — APPROVED** [review]
LGTM 🎉  - I'll make a new issue for those other concerns

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md [hunk 1]
```diff
@@ -88,7 +88,44 @@ var hash = string => {
...
+  this.add = function(key, val) {
+    var theHash = hash(key);
+    if (!this.collection.hasOwnProperty(theHash)) {
+      this.collection[theHash] = {};
+    }
+    this.collection[theHash][key] = val;
+  }
+
+  this.remove = function(key) {
+    var hashedObj = this.collection[hash(key)];
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md]:
```suggestion
    var theHash = hash(key);
    var hashedObj = this.collection[theHash];
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md [hunk 2]
```diff
@@ -88,7 +88,44 @@ var hash = string => {
...
+    if (hashedObj.hasOwnProperty(key)) {
+      delete hashedObj[key];
+    }
+    if (!Object.keys(hashedObj).length) {
+      delete this.collection[hash(key)];
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md]:
```suggestion
      delete this.collection[theHash];
```

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: 

If you feel like it's worth changing `remove` so that it's consistent with `add` and `lookup`, please do.  Either way, I think this is fine to merge.

**Pull request merged**